diff --git a/src/net/sourceforge/plantuml/classdiagram/command/CommandHideShow3.java b/src/net/sourceforge/plantuml/classdiagram/command/CommandHideShow3.java
index a3643b15..54234235 100644
--- a/src/net/sourceforge/plantuml/classdiagram/command/CommandHideShow3.java
+++ b/src/net/sourceforge/plantuml/classdiagram/command/CommandHideShow3.java
@@ -39,0 +40 @@ import java.util.StringTokenizer;
+import net.sourceforge.plantuml.StringUtils;
@@ -87 +88 @@ public class CommandHideShow3 extends SingleLineCommand2<UmlDiagram> {
-		final StringTokenizer st = new StringTokenizer(arg.get("VISIBILITY", 0).toLowerCase(), " ,");
+		final StringTokenizer st = new StringTokenizer(StringUtils.goLowerCase(arg.get("VISIBILITY", 0)), " ,");
@@ -125 +126 @@ public class CommandHideShow3 extends SingleLineCommand2<UmlDiagram> {
-		final String sub = s.substring(0, 3).toLowerCase();
+		final String sub = StringUtils.goLowerCase(s.substring(0, 3));