diff --git a/src/main/java/org/perfcake/parser/ScenarioParser.java b/src/main/java/org/perfcake/parser/ScenarioParser.java
index 4bcd86e3..8da6458b 100644
--- a/src/main/java/org/perfcake/parser/ScenarioParser.java
+++ b/src/main/java/org/perfcake/parser/ScenarioParser.java
@@ -23,0 +24 @@ import java.io.IOException;
+import java.io.UnsupportedEncodingException;
@@ -79 +80 @@ public class ScenarioParser {
-         Source scenarioXML = new StreamSource(new ByteArrayInputStream(scenarioConfig.getBytes()));
+         Source scenarioXML = new StreamSource(new ByteArrayInputStream(scenarioConfig.getBytes(Utils.getDefaultEncoding())));
@@ -92,0 +94,2 @@ public class ScenarioParser {
+      } catch (UnsupportedEncodingException e) {
+         throw new PerfCakeException("set encoding is not supported: ", e);