diff --git a/core/src/main/java/org/modelmapper/convention/InexactMatcher.java b/core/src/main/java/org/modelmapper/convention/InexactMatcher.java
index 19f2442..3c496e8 100644
--- a/core/src/main/java/org/modelmapper/convention/InexactMatcher.java
+++ b/core/src/main/java/org/modelmapper/convention/InexactMatcher.java
@@ -45 +45 @@ class InexactMatcher {
-      for (int srcIndex = srcStartIndex, dstIndex = dstStartIndex, srcCharIndex = 0, dstCharIndex = 0;;) {
+      for (int srcIndex = srcStartIndex, dstIndex = dstStartIndex, srcCharIndex = 0, dstCharIndex = 0; srcStr.length() > 0 && dstStr.length() > 0;) {
