diff --git a/server/src/main/java/com/orientechnologies/orient/server/distributed/task/OCreateRecordTask.java b/server/src/main/java/com/orientechnologies/orient/server/distributed/task/OCreateRecordTask.java
index 3976e2d..fe1f7fa 100644
--- a/server/src/main/java/com/orientechnologies/orient/server/distributed/task/OCreateRecordTask.java
+++ b/server/src/main/java/com/orientechnologies/orient/server/distributed/task/OCreateRecordTask.java
@@ -21,0 +22,4 @@ package com.orientechnologies.orient.server.distributed.task;
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+
@@ -38,4 +41,0 @@ import com.orientechnologies.orient.server.distributed.ODistributedServerManager
-import java.io.IOException;
-import java.io.ObjectInput;
-import java.io.ObjectOutput;
-
@@ -69,0 +70 @@ public class OCreateRecordTask extends OAbstractRecordReplicatedTask {
+      if (clazz != null)
