diff --git a/wicket-core/src/main/java/org/apache/wicket/ajax/json/JsonUtils.java b/wicket-core/src/main/java/org/apache/wicket/ajax/json/JsonUtils.java
index f4b8a9b..ece4997 100644
--- a/wicket-core/src/main/java/org/apache/wicket/ajax/json/JsonUtils.java
+++ b/wicket-core/src/main/java/org/apache/wicket/ajax/json/JsonUtils.java
@@ -61 +61,3 @@ public final class JsonUtils
-				else if (value.getClass().isArray())
+				else if (value != null)
+				{
+					if (value.getClass().isArray())
@@ -77,2 +78,0 @@ public final class JsonUtils
-					if (value != null)
-					{
