diff --git a/maven-core/src/main/java/org/apache/maven/plugin/internal/DefaultLegacySupport.java b/maven-core/src/main/java/org/apache/maven/plugin/internal/DefaultLegacySupport.java
index 81e69d9..dfd3f40 100644
--- a/maven-core/src/main/java/org/apache/maven/plugin/internal/DefaultLegacySupport.java
+++ b/maven-core/src/main/java/org/apache/maven/plugin/internal/DefaultLegacySupport.java
@@ -40 +40 @@
-    private ThreadLocal<MavenSession> session = new ThreadLocal<MavenSession>();
+    private ThreadLocal<MavenSession> session = new InheritableThreadLocal<MavenSession>();
