diff --git a/californium/src/main/java/ch/ethz/inf/vs/californium/coap/CoAP.java b/californium/src/main/java/ch/ethz/inf/vs/californium/coap/CoAP.java
index e6e03854..e17a63e8 100644
--- a/californium/src/main/java/ch/ethz/inf/vs/californium/coap/CoAP.java
+++ b/californium/src/main/java/ch/ethz/inf/vs/californium/coap/CoAP.java
@@ -16,2 +16,4 @@ public class CoAP {
-	public static String COAP_URI_SCHEME = "coap";
-	public static String COAP_SECURE_URI_SCHEME = "coaps";
+	public static final int VERSION = 0x01;
+	
+	public static final String COAP_URI_SCHEME = "coap";
+	public static final String COAP_SECURE_URI_SCHEME = "coaps";