diff --git a/processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java b/processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java
index 7835e06..26985ed 100644
--- a/processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java
+++ b/processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java
@@ -289,5 +288,0 @@ public class IncrementalIndex implements Iterable<Row>, Closeable
-    int index = 0;
-    for (String dim : incrementalIndexSchema.getDimensionsSpec().getDimensions()) {
-      dimensionOrder.put(dim, index++);
-      dimensions.add(dim);
-    }
