diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/DeleteTableProcedure.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/DeleteTableProcedure.java
index 4b95fa8..1e86254 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/DeleteTableProcedure.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/DeleteTableProcedure.java
@@ -34 +33,0 @@ import org.apache.hadoop.hbase.HRegionInfo;
-import org.apache.hadoop.hbase.HTableDescriptor;
@@ -348,5 +346,0 @@ public class DeleteTableProcedure
-    // Archive the mob data if there is a mob-enabled column
-    HTableDescriptor htd = env.getMasterServices().getTableDescriptors().get(tableName);
-    boolean hasMob = MobUtils.hasMobColumns(htd);
-    Path mobTableDir = null;
-    if (hasMob) {
@@ -354 +348 @@ public class DeleteTableProcedure
-      mobTableDir = FSUtils.getTableDir(new Path(mfs.getRootDir(), MobConstants.MOB_DIR_NAME),
+    Path mobTableDir = FSUtils.getTableDir(new Path(mfs.getRootDir(), MobConstants.MOB_DIR_NAME),
@@ -361 +354,0 @@ public class DeleteTableProcedure
-    }
@@ -369 +362 @@ public class DeleteTableProcedure
-    if (hasMob && mobTableDir != null && fs.exists(mobTableDir)) {
+    if (mobTableDir != null && fs.exists(mobTableDir)) {
