diff --git a/tika-parsers/src/main/java/org/apache/tika/parser/microsoft/WordExtractor.java b/tika-parsers/src/main/java/org/apache/tika/parser/microsoft/WordExtractor.java
index 15984fe..4c950fa 100644
--- a/tika-parsers/src/main/java/org/apache/tika/parser/microsoft/WordExtractor.java
+++ b/tika-parsers/src/main/java/org/apache/tika/parser/microsoft/WordExtractor.java
@@ -18,0 +19,2 @@ package org.apache.tika.parser.microsoft;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
@@ -56,2 +57,0 @@ import org.xml.sax.helpers.AttributesImpl;
-import static java.nio.charset.StandardCharsets.UTF_8;
-
@@ -311 +311,6 @@ public class WordExtractor extends AbstractPOIFSExtractor {
-                    String id = "_" + field.getMarkSeparatorCharacterRun(r).getPicOffset();
+                    String id = "_unknown_id";
+                    //this can return null (TIKA-1956)
+                    CharacterRun mscr = field.getMarkSeparatorCharacterRun(r);
+                    if (mscr != null) {
+                        id = "_" + mscr.getPicOffset();
+                    }
