diff --git a/API/src/main/java/org/sikuli/util/LinuxSupport.java b/API/src/main/java/org/sikuli/util/LinuxSupport.java
index 839c367..2c5c4d4 100644
--- a/API/src/main/java/org/sikuli/util/LinuxSupport.java
+++ b/API/src/main/java/org/sikuli/util/LinuxSupport.java
@@ -287 +287 @@ public class LinuxSupport {
-    File[] javas = new File[2];
+    File[] javas = new File[] {null, null} ;
@@ -289 +289,4 @@ public class LinuxSupport {
-    javas[1] = new File(System.getenv("JAVA_HOME"));
+    String jhome = System.getenv("JAVA_HOME");
+    if (jhome != null) {
+      javas[1] = new File(jhome);
+    }
