diff --git a/src/main/java/com/atlauncher/nio/JsonFile.java b/src/main/java/com/atlauncher/nio/JsonFile.java
index 12257c2b..30e57d6d 100644
--- a/src/main/java/com/atlauncher/nio/JsonFile.java
+++ b/src/main/java/com/atlauncher/nio/JsonFile.java
@@ -67 +67 @@ public final class JsonFile {
-        OutputStreamWriter osw = new OutputStreamWriter(os);
+        OutputStreamWriter osw = new OutputStreamWriter(os, "UTF-8");