diff --git a/src/main/java/com/salesforce/hbase/index/write/ParallelWriterIndexCommitter.java b/src/main/java/com/salesforce/hbase/index/write/ParallelWriterIndexCommitter.java
index aadd5a19..4adbcb12 100644
--- a/src/main/java/com/salesforce/hbase/index/write/ParallelWriterIndexCommitter.java
+++ b/src/main/java/com/salesforce/hbase/index/write/ParallelWriterIndexCommitter.java
@@ -71 +71 @@ public class ParallelWriterIndexCommitter implements IndexCommitter {
-  public static String NUM_CONCURRENT_INDEX_WRITER_THREADS_CONF_KEY = "index.writer.threads.max";
+  public static final String NUM_CONCURRENT_INDEX_WRITER_THREADS_CONF_KEY = "index.writer.threads.max";