diff --git a/modules/activiti-engine/src/main/java/org/activiti/engine/impl/bpmn/diagram/ProcessDiagramCanvas.java b/modules/activiti-engine/src/main/java/org/activiti/engine/impl/bpmn/diagram/ProcessDiagramCanvas.java
index 72d7414..32b876d 100644
--- a/modules/activiti-engine/src/main/java/org/activiti/engine/impl/bpmn/diagram/ProcessDiagramCanvas.java
+++ b/modules/activiti-engine/src/main/java/org/activiti/engine/impl/bpmn/diagram/ProcessDiagramCanvas.java
@@ -341,2 +341,6 @@ public class ProcessDiagramCanvas {
-    while (fontMetrics.stringWidth(text + "...") > (width - 10)) {
-      text = text.substring(0, text.length() - 2);
+
+    // remove length for "..."
+    int maxWidth = width - 10;
+    
+    while (fontMetrics.stringWidth(text + "...") > maxWidth && text.length()>0) {
+      text = text.substring(0, text.length() - 1);
