diff --git a/src/main/java/com/googlecode/objectify/util/jackson/ObjectifyJacksonModule.java b/src/main/java/com/googlecode/objectify/util/jackson/ObjectifyJacksonModule.java
index a8e19f6e..d2e305de 100644
--- a/src/main/java/com/googlecode/objectify/util/jackson/ObjectifyJacksonModule.java
+++ b/src/main/java/com/googlecode/objectify/util/jackson/ObjectifyJacksonModule.java
@@ -13,0 +14 @@ public class ObjectifyJacksonModule extends SimpleModule {
+	private static final long serialVersionUID = 1L;
@@ -32 +33 @@ public class ObjectifyJacksonModule extends SimpleModule {
-		
+