diff --git a/src/net/sourceforge/plantuml/command/CommandMultilinesFooter.java b/src/net/sourceforge/plantuml/command/CommandMultilinesFooter.java
index eca94312..910c9559 100644
--- a/src/net/sourceforge/plantuml/command/CommandMultilinesFooter.java
+++ b/src/net/sourceforge/plantuml/command/CommandMultilinesFooter.java
@@ -31 +31 @@
- * Revision $Revision: 14321 $
+ * Revision $Revision: 14726 $
@@ -63 +63 @@ public class CommandMultilinesFooter extends CommandMultilines<UmlDiagram> {
-			diagram.setFooterAlignment(HorizontalAlignment.valueOf(align.toUpperCase()));
+			diagram.setFooterAlignment(HorizontalAlignment.valueOf(StringUtils.goUpperCase(align)));