diff --git a/src/main/java/com/ldbc/driver/util/CsvFileReader.java b/src/main/java/com/ldbc/driver/util/CsvFileReader.java
index 2d2c4133..112f098c 100644
--- a/src/main/java/com/ldbc/driver/util/CsvFileReader.java
+++ b/src/main/java/com/ldbc/driver/util/CsvFileReader.java
@@ -2,0 +3,2 @@ package com.ldbc.driver.util;
+import com.google.common.base.Charsets;
+
@@ -9,2 +10,0 @@ public class CsvFileReader implements Iterator<String[]> {
-    public static final String CSV_SEPARATOR = "\\|";
-
@@ -18 +18 @@ public class CsvFileReader implements Iterator<String[]> {
-        this.csvReader = new BufferedReader(new FileReader(csvFile));
+        this.csvReader = new BufferedReader(new InputStreamReader(new FileInputStream(csvFile), Charsets.UTF_8));