diff --git a/src/main/java/com/graphhopper/storage/Location2IDPreciseIndex.java b/src/main/java/com/graphhopper/storage/Location2IDPreciseIndex.java
index 9d519ba3d..640fb60c7 100644
--- a/src/main/java/com/graphhopper/storage/Location2IDPreciseIndex.java
+++ b/src/main/java/com/graphhopper/storage/Location2IDPreciseIndex.java
@@ -25,0 +26 @@ import com.graphhopper.util.EdgeIterator;
+import com.graphhopper.util.NumHelper;
@@ -52 +52,0 @@ public class Location2IDPreciseIndex implements Location2IDIndex {
-    private Logger logger = LoggerFactory.getLogger(getClass());
@@ -199 +199,2 @@ public class Location2IDPreciseIndex implements Location2IDIndex {
-                                if (tryLon == tmpLon && tryLat == tmpLat || tryLon == connLon && tryLat == connLat)
+                                if (NumHelper.equals(tryLon, tmpLon) && NumHelper.equals(tryLat, tmpLat)
+                                        || NumHelper.equals(tryLon, connLon) && NumHelper.equals(tryLat, connLat))