diff --git a/rio/src/main/java/org/semanticweb/owlapi/rio/RioN3OntologyStorerFactory.java b/rio/src/main/java/org/semanticweb/owlapi/rio/RioN3OntologyStorerFactory.java
index e32f5355b..5c0fd18b7 100644
--- a/rio/src/main/java/org/semanticweb/owlapi/rio/RioN3OntologyStorerFactory.java
+++ b/rio/src/main/java/org/semanticweb/owlapi/rio/RioN3OntologyStorerFactory.java
@@ -50,0 +51,2 @@ public class RioN3OntologyStorerFactory implements RioRDFOntologyStorerFactory {
+    private static final long serialVersionUID = 40000L;
+