diff --git a/src/net/sourceforge/plantuml/classdiagram/command/CommandCreateClassMultilines.java b/src/net/sourceforge/plantuml/classdiagram/command/CommandCreateClassMultilines.java
index 81bf3e4d..cd394825 100644
--- a/src/net/sourceforge/plantuml/classdiagram/command/CommandCreateClassMultilines.java
+++ b/src/net/sourceforge/plantuml/classdiagram/command/CommandCreateClassMultilines.java
@@ -173 +173 @@ public class CommandCreateClassMultilines extends CommandMultilines2<ClassDiagra
-		final LeafType type = LeafType.getLeafType(arg.get("TYPE", 0).toUpperCase());
+		final LeafType type = LeafType.getLeafType(StringUtils.goUpperCase(arg.get("TYPE", 0)));
@@ -228 +228 @@ public class CommandCreateClassMultilines extends CommandMultilines2<ClassDiagra
-		final LinkStyle style = LinkStyle.valueOf(s.toUpperCase());
+		final LinkStyle style = LinkStyle.valueOf(StringUtils.goUpperCase(s));