diff --git a/src/main/java/fr/ens/transcriptome/eoulsan/bio/readsmappers/SimpleGenomeIndexStorage.java b/src/main/java/fr/ens/transcriptome/eoulsan/bio/readsmappers/SimpleGenomeIndexStorage.java
index d7c00da0..075cd68f 100644
--- a/src/main/java/fr/ens/transcriptome/eoulsan/bio/readsmappers/SimpleGenomeIndexStorage.java
+++ b/src/main/java/fr/ens/transcriptome/eoulsan/bio/readsmappers/SimpleGenomeIndexStorage.java
@@ -137 +137,2 @@ public class SimpleGenomeIndexStorage implements GenomeIndexStorage {
-        new BufferedReader(new InputStreamReader(indexFile.open()));
+        new BufferedReader(new InputStreamReader(indexFile.open(),
+            Globals.DEFAULT_FILE_ENCODING));
@@ -180 +181,2 @@ public class SimpleGenomeIndexStorage implements GenomeIndexStorage {
-        new BufferedWriter(new OutputStreamWriter(indexFile.create()));
+        new BufferedWriter(new OutputStreamWriter(indexFile.create(),
+            Globals.DEFAULT_FILE_ENCODING));