diff --git a/sjk-core/src/main/java/org/gridkit/jvmtool/MBeanCpuUsageReporter.java b/sjk-core/src/main/java/org/gridkit/jvmtool/MBeanCpuUsageReporter.java
index 7ac19f6..8070176 100644
--- a/sjk-core/src/main/java/org/gridkit/jvmtool/MBeanCpuUsageReporter.java
+++ b/sjk-core/src/main/java/org/gridkit/jvmtool/MBeanCpuUsageReporter.java
@@ -186,0 +187,3 @@ public class MBeanCpuUsageReporter {
+    	            if (threadDump.get(ids[i]) == null) {
+    	                continue;
+    	            }
@@ -194,0 +198,3 @@ public class MBeanCpuUsageReporter {
+                    if (threadDump.get(ids[i]) == null) {
+                        continue;
+                    }
@@ -200,0 +207,3 @@ public class MBeanCpuUsageReporter {
+                    if (threadDump.get(id) == null) {
+                        continue;
+                    }
