diff --git a/modules/activiti-engine/src/main/java/org/activiti/engine/impl/bpmn/behavior/ShellActivityBehavior.java b/modules/activiti-engine/src/main/java/org/activiti/engine/impl/bpmn/behavior/ShellActivityBehavior.java
index 8ae2688d8..2f5175b07 100644
--- a/modules/activiti-engine/src/main/java/org/activiti/engine/impl/bpmn/behavior/ShellActivityBehavior.java
+++ b/modules/activiti-engine/src/main/java/org/activiti/engine/impl/bpmn/behavior/ShellActivityBehavior.java
@@ -18 +17,0 @@ import org.activiti.engine.delegate.Expression;
-import org.activiti.engine.impl.delegate.ActivityExecution;
@@ -21,0 +21,2 @@ public class ShellActivityBehavior extends AbstractBpmnActivityBehavior {
+  private static final long serialVersionUID = 1L;
+  
@@ -49 +50 @@ public class ShellActivityBehavior extends AbstractBpmnActivityBehavior {
-  private void readFields(ActivityExecution execution) {
+  private void readFields(DelegateExecution execution) {
@@ -70 +71 @@ public class ShellActivityBehavior extends AbstractBpmnActivityBehavior {
-  public void execute(ActivityExecution execution) {
+  public void execute(DelegateExecution execution) {