diff --git a/src/main/java/org/codesecure/dependencycheck/dependency/Dependency.java b/src/main/java/org/codesecure/dependencycheck/dependency/Dependency.java
index 6053026e1..579cd1385 100644
--- a/src/main/java/org/codesecure/dependencycheck/dependency/Dependency.java
+++ b/src/main/java/org/codesecure/dependencycheck/dependency/Dependency.java
@@ -379,2 +378,0 @@ public class Dependency {
-        String fnd = str.toLowerCase();
-
@@ -387 +385 @@ public class Dependency {
-        if (versionEvidence.containsUsedString(fnd)) {
+        if (versionEvidence.containsUsedString(str)) {