diff --git a/src/main/java/org/perfcake/reporting/reporters/MemoryUsageReporter.java b/src/main/java/org/perfcake/reporting/reporters/MemoryUsageReporter.java
index 61fe4002..d31611c6 100644
--- a/src/main/java/org/perfcake/reporting/reporters/MemoryUsageReporter.java
+++ b/src/main/java/org/perfcake/reporting/reporters/MemoryUsageReporter.java
@@ -24,0 +25 @@ import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
@@ -37,0 +39 @@ import org.perfcake.reporting.reporters.accumulators.LastValueAccumulator;
+import org.perfcake.util.Utils;
@@ -133,2 +135,2 @@ public class MemoryUsageReporter extends AbstractReporter {
-         requestWriter = new PrintWriter(socket.getOutputStream(), true);
-         responseReader = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+         requestWriter = new PrintWriter(new OutputStreamWriter(socket.getOutputStream(), Utils.getDefaultEncoding()), true);
+         responseReader = new BufferedReader(new InputStreamReader(socket.getInputStream(), Utils.getDefaultEncoding()));