diff --git a/src/main/java/com/graphhopper/util/CmdArgs.java b/src/main/java/com/graphhopper/util/CmdArgs.java
index 778d25d26..a4b22c9d2 100644
--- a/src/main/java/com/graphhopper/util/CmdArgs.java
+++ b/src/main/java/com/graphhopper/util/CmdArgs.java
@@ -17,0 +18 @@ package com.graphhopper.util;
+import java.io.FileInputStream;
@@ -19,0 +21 @@ import java.io.IOException;
+import java.io.InputStreamReader;
@@ -105 +107 @@ public class CmdArgs {
-        Helper.loadProperties(map, new FileReader(fileStr));
+        Helper.loadProperties(map, new InputStreamReader(new FileInputStream(fileStr), "UTF-8"));