diff --git a/webmagic-core/src/main/java/us/codecraft/webmagic/downloader/HttpClientPool.java b/webmagic-core/src/main/java/us/codecraft/webmagic/downloader/HttpClientPool.java
index c256ac4..f3b72b6 100644
--- a/webmagic-core/src/main/java/us/codecraft/webmagic/downloader/HttpClientPool.java
+++ b/webmagic-core/src/main/java/us/codecraft/webmagic/downloader/HttpClientPool.java
@@ -54 +53,0 @@ public class HttpClientPool {
-        }
@@ -56,0 +56,6 @@ public class HttpClientPool {
+        } else {
+            params.setIntParameter(CoreConnectionPNames.SO_TIMEOUT, 3000);
+            params.setIntParameter(CoreConnectionPNames.CONNECTION_TIMEOUT, 3000);
+        }
+
+
@@ -76 +80,0 @@ public class HttpClientPool {
-
