diff --git a/processor/src/main/java/org/mapstruct/ap/MapperGenerationVisitor.java b/processor/src/main/java/org/mapstruct/ap/MapperGenerationVisitor.java
index b958749..2f38a5e 100644
--- a/processor/src/main/java/org/mapstruct/ap/MapperGenerationVisitor.java
+++ b/processor/src/main/java/org/mapstruct/ap/MapperGenerationVisitor.java
@@ -465,0 +466,8 @@ private Method getReverseMappingMethod(List<Method> rawMethods,
+                    ExecutableElement correspondingSetter = Executables.getCorrespondingPropertyAccessor(
+                        getterMethod,
+                        sourceSetters
+                    );
+                    ExecutableElement correspondingGetter = Executables.getCorrespondingPropertyAccessor(
+                        setterMethod,
+                        targetGetters
+                    );
@@ -470,3 +478 @@ private Method getReverseMappingMethod(List<Method> rawMethods,
-                            Executables.getCorrespondingPropertyAccessor( getterMethod, sourceSetters )
-                                .getSimpleName()
-                                .toString(),
+                            correspondingSetter != null ? correspondingSetter.getSimpleName().toString() : null,
@@ -475,3 +481 @@ private Method getReverseMappingMethod(List<Method> rawMethods,
-                            Executables.getCorrespondingPropertyAccessor( setterMethod, targetGetters )
-                                .getSimpleName()
-                                .toString(),
+                            correspondingGetter != null ? correspondingGetter.getSimpleName().toString() : null,
