diff --git a/module-client/src/main/java/org/apache/http/impl/client/AbstractAuthenticationHandler.java b/module-client/src/main/java/org/apache/http/impl/client/AbstractAuthenticationHandler.java
index 4c114fdad..88c83b0c8 100644
--- a/module-client/src/main/java/org/apache/http/impl/client/AbstractAuthenticationHandler.java
+++ b/module-client/src/main/java/org/apache/http/impl/client/AbstractAuthenticationHandler.java
@@ -37,0 +38 @@ import java.util.List;
+import java.util.Locale;
@@ -100 +101 @@ public abstract class AbstractAuthenticationHandler implements AuthenticationHan
-            map.put(s.toLowerCase(), header);
+            map.put(s.toLowerCase(Locale.ENGLISH), header);
@@ -129 +130 @@ public abstract class AbstractAuthenticationHandler implements AuthenticationHan
-            Header challenge = challenges.get(id.toLowerCase()); 
+            Header challenge = challenges.get(id.toLowerCase(Locale.ENGLISH));