diff --git a/src/main/java/org/apache/commons/lang3/RandomStringUtils.java b/src/main/java/org/apache/commons/lang3/RandomStringUtils.java
index bded154..a799057 100644
--- a/src/main/java/org/apache/commons/lang3/RandomStringUtils.java
+++ b/src/main/java/org/apache/commons/lang3/RandomStringUtils.java
@@ -214 +214 @@ public static String random(int count, int start, int end, boolean letters, bool
-     * @param chars  the set of chars to choose randoms from.
+     * @param chars  the set of chars to choose randoms from, must not be empty.
@@ -220 +220 @@ public static String random(int count, int start, int end, boolean letters, bool
-     * @throws IllegalArgumentException if {@code count} &lt; 0.
+     * @throws IllegalArgumentException if {@code count} &lt; 0 or the provided chars array is empty.
@@ -229,0 +230,4 @@ public static String random(int count, int start, int end, boolean letters, bool
+        if (chars != null && chars.length == 0) {
+            throw new IllegalArgumentException("The chars array must not be empty");
+        }
+
@@ -231,2 +235,3 @@ public static String random(int count, int start, int end, boolean letters, bool
-            end = 'z' + 1;
-            start = ' ';
+            if (chars != null) {
+                end = chars.length;
+            } else {
@@ -234 +238,0 @@ public static String random(int count, int start, int end, boolean letters, bool
-                start = 0;
@@ -235,0 +240,4 @@ public static String random(int count, int start, int end, boolean letters, bool
+                } else {
+                    end = 'z' + 1;
+                    start = ' ';                
+                }
@@ -288 +296,2 @@ public static String random(int count, int start, int end, boolean letters, bool
-     * specified.</p>
+     * specified by the string, must not be empty. 
+     * If null, the set of all characters is used.</p>
@@ -292 +301 @@ public static String random(int count, int start, int end, boolean letters, bool
-     *  may be null
+     *  may be null, but must not be empty
@@ -294 +303 @@ public static String random(int count, int start, int end, boolean letters, bool
-     * @throws IllegalArgumentException if {@code count} &lt; 0.
+     * @throws IllegalArgumentException if {@code count} &lt; 0 or the string is empty.
