diff --git a/maven-core/src/main/java/org/apache/maven/DefaultMaven.java b/maven-core/src/main/java/org/apache/maven/DefaultMaven.java
index 6328819..ab47efd 100644
--- a/maven-core/src/main/java/org/apache/maven/DefaultMaven.java
+++ b/maven-core/src/main/java/org/apache/maven/DefaultMaven.java
@@ -270,2 +269,0 @@ private MavenExecutionResult doExecute( MavenExecutionRequest request )
-        session.setProjects( projectDependencyGraph.getSortedProjects() );
-        
@@ -276,0 +275,2 @@ private MavenExecutionResult doExecute( MavenExecutionRequest request )
+        session.setProjects( projectDependencyGraph.getSortedProjects() );
+
