diff --git a/module-client/src/main/java/org/apache/http/auth/AuthScope.java b/module-client/src/main/java/org/apache/http/auth/AuthScope.java
index c8cf43052..344d6ff1c 100644
--- a/module-client/src/main/java/org/apache/http/auth/AuthScope.java
+++ b/module-client/src/main/java/org/apache/http/auth/AuthScope.java
@@ -32,0 +33,2 @@ package org.apache.http.auth;
+import java.util.Locale;
+
@@ -107 +109 @@ public class AuthScope {
-        this.host =   (host == null)   ? ANY_HOST: host.toLowerCase();
+        this.host =   (host == null)   ? ANY_HOST: host.toLowerCase(Locale.ENGLISH);
@@ -110 +112 @@ public class AuthScope {
-        this.scheme = (scheme == null) ? ANY_SCHEME: scheme.toUpperCase();
+        this.scheme = (scheme == null) ? ANY_SCHEME: scheme.toUpperCase(Locale.ENGLISH);
@@ -257 +259 @@ public class AuthScope {
-            buffer.append(this.scheme.toUpperCase());
+            buffer.append(this.scheme.toUpperCase(Locale.ENGLISH));