diff --git a/src/net/sourceforge/plantuml/cute/CuteShapeFactory.java b/src/net/sourceforge/plantuml/cute/CuteShapeFactory.java
index ab68e29b..abee1c00 100644
--- a/src/net/sourceforge/plantuml/cute/CuteShapeFactory.java
+++ b/src/net/sourceforge/plantuml/cute/CuteShapeFactory.java
@@ -37,0 +38,2 @@ import java.util.Map;
+import net.sourceforge.plantuml.StringUtils;
+
@@ -53 +55 @@ public class CuteShapeFactory {
-		data = data.toLowerCase().trim();
+		data = StringUtils.trin(data.toLowerCase());