diff --git a/util/src/main/java/com/psddev/dari/util/TypeDefinition.java b/util/src/main/java/com/psddev/dari/util/TypeDefinition.java
index 24e280c2..2141d3c2 100644
--- a/util/src/main/java/com/psddev/dari/util/TypeDefinition.java
+++ b/util/src/main/java/com/psddev/dari/util/TypeDefinition.java
@@ -14,0 +15 @@ import java.util.List;
+import java.util.Locale;
@@ -372 +373 @@ public class TypeDefinition<T> {
-                                    ? nameMatcher.group(3).toLowerCase() + nameMatcher.group(4)
+                                    ? nameMatcher.group(3).toLowerCase(Locale.ENGLISH) + nameMatcher.group(4)
@@ -412 +413 @@ public class TypeDefinition<T> {
-                            setters.put(nameMatcher.group(1).toLowerCase() + nameMatcher.group(2), method);
+                            setters.put(nameMatcher.group(1).toLowerCase(Locale.ENGLISH) + nameMatcher.group(2), method);