diff --git a/job/src/main/java/com/kylinolap/job/cmd/JavaHadoopCmdOutput.java b/job/src/main/java/com/kylinolap/job/cmd/JavaHadoopCmdOutput.java
index 20dd9dd..c3f9236 100644
--- a/job/src/main/java/com/kylinolap/job/cmd/JavaHadoopCmdOutput.java
+++ b/job/src/main/java/com/kylinolap/job/cmd/JavaHadoopCmdOutput.java
@@ -20,0 +21 @@ import java.util.Map;
+import org.apache.hadoop.mapreduce.Counters;
@@ -162,2 +163,9 @@ public class JavaHadoopCmdOutput implements ICommandOutput {
-            this.output.append(job.getCounters().toString()).append("\n");
-            log.debug(job.getCounters().toString());
+            Counters counters = job.getCounters();
+            if (counters == null) {
+                String errorMsg = "no counters for job " + mrJobID;
+                log.warn(errorMsg);
+                output.append(errorMsg);
+                return;
+            }
+            this.output.append(counters.toString()).append("\n");
+            log.debug(counters.toString());
@@ -169 +177 @@ public class JavaHadoopCmdOutput implements ICommandOutput {
-            long mapInputRecords = job.getCounters().findCounter(TaskCounter.MAP_INPUT_RECORDS).getValue();
+            long mapInputRecords = counters.findCounter(TaskCounter.MAP_INPUT_RECORDS).getValue();
@@ -171 +179 @@ public class JavaHadoopCmdOutput implements ICommandOutput {
-            long hdfsBytesWritten = job.getCounters().findCounter("FileSystemCounters", "HDFS_BYTES_WRITTEN").getValue();
+            long hdfsBytesWritten = counters.findCounter("FileSystemCounters", "HDFS_BYTES_WRITTEN").getValue();
