diff --git a/API/src/main/java/org/sikuli/script/Region.java b/API/src/main/java/org/sikuli/script/Region.java
index 47aeefa..f4227e0 100644
--- a/API/src/main/java/org/sikuli/script/Region.java
+++ b/API/src/main/java/org/sikuli/script/Region.java
@@ -2707,0 +2708,4 @@ public class Region {
+    double findTimeout = autoWaitTimeout; 
+    if (repeating != null) {
+      findTimeout = repeating.getFindTimeOut();
+    }
@@ -2729 +2733 @@ public class Region {
-            f = checkLastSeenAndCreateFinder(img, repeating.getFindTimeOut(), null);
+            f = checkLastSeenAndCreateFinder(img, findTimeout, null);
@@ -2752 +2756 @@ public class Region {
-          f = checkLastSeenAndCreateFinder(img, repeating.getFindTimeOut(), (Pattern) ptn);
+          f = checkLastSeenAndCreateFinder(img, findTimeout, (Pattern) ptn);
@@ -2764 +2768 @@ public class Region {
-          f = checkLastSeenAndCreateFinder(img, repeating.getFindTimeOut(), null);
+          f = checkLastSeenAndCreateFinder(img, findTimeout, null);
