diff --git a/util/src/main/java/com/psddev/dari/util/ObjectUtils.java b/util/src/main/java/com/psddev/dari/util/ObjectUtils.java
index 41e929b4..b5c8291c 100644
--- a/util/src/main/java/com/psddev/dari/util/ObjectUtils.java
+++ b/util/src/main/java/com/psddev/dari/util/ObjectUtils.java
@@ -18,0 +19 @@ import java.util.List;
+import java.util.Locale;
@@ -392 +393 @@ public abstract class ObjectUtils {
-                        String contentType = items[0].toLowerCase();
+                        String contentType = items[0].toLowerCase(Locale.ENGLISH);
@@ -395 +396 @@ public abstract class ObjectUtils {
-                            contentTypes.put(items[j].toLowerCase(), contentType);
+                            contentTypes.put(items[j].toLowerCase(Locale.ENGLISH), contentType);
@@ -416 +417 @@ public abstract class ObjectUtils {
-            String type = CONTENT_TYPES.get().get(fileName.substring(dotAt + 1).toLowerCase());
+            String type = CONTENT_TYPES.get().get(fileName.substring(dotAt + 1).toLowerCase(Locale.ENGLISH));
@@ -600 +601 @@ public abstract class ObjectUtils {
-                string.substring(0, 1).toUpperCase() + string.substring(1);
+                string.substring(0, 1).toUpperCase(Locale.ENGLISH) + string.substring(1);