diff --git a/processing/src/main/java/io/druid/query/topn/TopNBinaryFn.java b/processing/src/main/java/io/druid/query/topn/TopNBinaryFn.java
index 18d7fa0..fd2b7a2 100644
--- a/processing/src/main/java/io/druid/query/topn/TopNBinaryFn.java
+++ b/processing/src/main/java/io/druid/query/topn/TopNBinaryFn.java
@@ -94 +94,13 @@ public class TopNBinaryFn implements BinaryFn<Result<TopNResultValue>, Result<To
-      retVals.put(arg1Val.getStringDimensionValue(dimension), arg1Val);
+      final String dimensionValue = arg1Val.getStringDimensionValue(dimension);
+      Map<String, Object> retVal = new LinkedHashMap<String, Object>(aggregations.size() + 2);
+      retVal.put(dimension, dimensionValue);
+
+      for (AggregatorFactory factory : aggregations) {
+        final String metricName = factory.getName();
+        retVal.put(metricName, arg1Val.getMetric(metricName));
+      }
+      for (PostAggregator postAgg : postAggregations) {
+        retVal.put(postAgg.getName(), postAgg.compute(retVal));
+      }
+
+      retVals.put(dimensionValue, new DimensionAndMetricValueExtractor(retVal));
@@ -115 +127,12 @@ public class TopNBinaryFn implements BinaryFn<Result<TopNResultValue>, Result<To
-        retVals.put(dimensionValue, arg2Val);
+        Map<String, Object> retVal = new LinkedHashMap<String, Object>(aggregations.size() + 2);
+        retVal.put(dimension, dimensionValue);
+
+        for (AggregatorFactory factory : aggregations) {
+          final String metricName = factory.getName();
+          retVal.put(metricName, arg2Val.getMetric(metricName));
+        }
+        for (PostAggregator postAgg : postAggregations) {
+          retVal.put(postAgg.getName(), postAgg.compute(retVal));
+        }
+
+        retVals.put(dimensionValue, new DimensionAndMetricValueExtractor(retVal));
