diff --git a/src/main/java/org/perfcake/validation/RulesBuilder.java b/src/main/java/org/perfcake/validation/RulesBuilder.java
index 083931cd..8dec52eb 100644
--- a/src/main/java/org/perfcake/validation/RulesBuilder.java
+++ b/src/main/java/org/perfcake/validation/RulesBuilder.java
@@ -31,0 +32 @@ import org.drools.rule.Package;
+import org.perfcake.util.Utils;
@@ -80 +81 @@ public class RulesBuilder {
-            final String rule = String.format("rule \"Line %d\"\n  dialect \"java\"\n  when\n    %s\n  then\n   > rulesUsed.remove(%d);\nend\n", lineNumber, assertion, lineNumber - 1, lineNumber);// rules
+            final String rule = String.format("rule \"Line %d\"%n  dialect \"java\"%n  when%n    %s%n  then%n   > rulesUsed.remove(%d);%nend%n", lineNumber, assertion, lineNumber - 1, lineNumber);// rules
@@ -94 +95 @@ public class RulesBuilder {
-      pBuilder.addPackageFromDrl(new StringReader(sBuilder.toString()), new InputStreamReader(dslis));
+      pBuilder.addPackageFromDrl(new StringReader(sBuilder.toString()), new InputStreamReader(dslis, Utils.getDefaultEncoding()));