diff --git a/src/net/sourceforge/plantuml/asciiart/ComponentTextGroupingHeader.java b/src/net/sourceforge/plantuml/asciiart/ComponentTextGroupingHeader.java
index fd01e9a2..3e2addc4 100644
--- a/src/net/sourceforge/plantuml/asciiart/ComponentTextGroupingHeader.java
+++ b/src/net/sourceforge/plantuml/asciiart/ComponentTextGroupingHeader.java
@@ -73 +73 @@ public class ComponentTextGroupingHeader extends AbstractComponentText {
-			charArea.drawStringLR(text.toUpperCase() + "  /", 2, 1);
+			charArea.drawStringLR(StringUtils.goUpperCase(text) + "  /", 2, 1);
@@ -86 +86 @@ public class ComponentTextGroupingHeader extends AbstractComponentText {
-			charArea.drawStringLR(text.toUpperCase() + "  /", 2, 1);
+			charArea.drawStringLR(StringUtils.goUpperCase(text) + "  /", 2, 1);