diff --git a/biojava-structure/src/main/java/org/biojava/nbio/structure/SeqMisMatchImpl.java b/biojava-structure/src/main/java/org/biojava/nbio/structure/SeqMisMatchImpl.java
index cb2cfa7ad..34f861cf6 100644
--- a/biojava-structure/src/main/java/org/biojava/nbio/structure/SeqMisMatchImpl.java
+++ b/biojava-structure/src/main/java/org/biojava/nbio/structure/SeqMisMatchImpl.java
@@ -10 +10,3 @@ public class SeqMisMatchImpl implements SeqMisMatch, Serializable{
-    Integer seqNum;
+	private static final long serialVersionUID = -3699285122925652562L;
+	
+	Integer seqNum;