diff --git a/src/main/java/io/mycat/route/handler/HintSchemaHandler.java b/src/main/java/io/mycat/route/handler/HintSchemaHandler.java
index 5430fdf..66214a3 100644
--- a/src/main/java/io/mycat/route/handler/HintSchemaHandler.java
+++ b/src/main/java/io/mycat/route/handler/HintSchemaHandler.java
@@ -49,2 +49,2 @@ public class HintSchemaHandler implements HintHandler {
-	    SchemaConfig tempSchema = MycatServer.getInstance().getConfig().getSchemas()
-				.get(hintSQLValue);
+		
+	    SchemaConfig tempSchema = MycatServer.getInstance().getConfig().getSchemas().get(hintSQLValue);
@@ -52,2 +52 @@ public class HintSchemaHandler implements HintHandler {
-			return routeStrategy.route(sysConfig, tempSchema,
-					sqlType, realSQL, charset, sc, cachePool);
+			return routeStrategy.route(sysConfig, tempSchema, sqlType, realSQL, charset, sc, cachePool);
@@ -55 +54 @@ public class HintSchemaHandler implements HintHandler {
-			String msg = "can't find schema:" + tempSchema.getName();
+			String msg = "can't find hint schema:" + hintSQLValue;
