diff --git a/logback-classic/src/main/java/ch/qos/logback/classic/spi/PackagingDataCalculator.java b/logback-classic/src/main/java/ch/qos/logback/classic/spi/PackagingDataCalculator.java
index e2f6dd4..4b89ac8 100644
--- a/logback-classic/src/main/java/ch/qos/logback/classic/spi/PackagingDataCalculator.java
+++ b/logback-classic/src/main/java/ch/qos/logback/classic/spi/PackagingDataCalculator.java
@@ -16,0 +17 @@ import java.net.URL;
+import java.security.CodeSource;
@@ -165 +166,3 @@ public class PackagingDataCalculator {
-        URL resource = type.getProtectionDomain().getCodeSource().getLocation();
+        CodeSource codeSource = type.getProtectionDomain().getCodeSource();
+        if (codeSource != null) {
+          URL resource = codeSource.getLocation();
@@ -175,0 +179 @@ public class PackagingDataCalculator {
+      }
