diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/NewHandler.java b/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/NewHandler.java
index 7a6566c..56f7742 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/NewHandler.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/NewHandler.java
@@ -26 +25,0 @@ import com.puppycrawl.tools.checkstyle.api.TokenTypes;
- *
@@ -27,0 +27 @@ import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+ * @author Ilja Dubinin
@@ -49,0 +50 @@ public class NewHandler extends ExpressionHandler
+        if (type != null) {
@@ -50,0 +52 @@ public class NewHandler extends ExpressionHandler
+        }
