diff --git a/structr-ui/src/main/java/org/structr/websocket/command/SearchCommand.java b/structr-ui/src/main/java/org/structr/websocket/command/SearchCommand.java
index 59b675a..b3d893d 100644
--- a/structr-ui/src/main/java/org/structr/websocket/command/SearchCommand.java
+++ b/structr-ui/src/main/java/org/structr/websocket/command/SearchCommand.java
@@ -65 +65,5 @@ public class SearchCommand extends AbstractCommand {
-		Class type                             = EntityContext.getEntityClassForRawType(typeString);
+		
+		Class type = null;
+		if (typeString != null) {
+			type = EntityContext.getEntityClassForRawType(typeString);
+		}
