diff --git a/src/main/java/org/perfcake/reporting/destinations/chart/Chart.java b/src/main/java/org/perfcake/reporting/destinations/chart/Chart.java
index b7fc8b96..068444bd 100644
--- a/src/main/java/org/perfcake/reporting/destinations/chart/Chart.java
+++ b/src/main/java/org/perfcake/reporting/destinations/chart/Chart.java
@@ -193 +193 @@ public class Chart {
-      final String loaderEntry = new String(Files.readAllBytes(Paths.get(descriptionFile.toURI())));
+      final String loaderEntry = new String(Files.readAllBytes(Paths.get(descriptionFile.toURI())), Utils.getDefaultEncoding());
@@ -208,0 +209,4 @@ public class Chart {
+      if (firstDataLine == null) {
+         throw new IOException(String.format("Cannot read chart from file %s. No entries in the data file.", descriptionFile.getAbsolutePath()));
+      }
+