diff --git a/src/main/java/com/github/sommeri/less4j/core/compiler/stages/SimpleImportsSolver.java b/src/main/java/com/github/sommeri/less4j/core/compiler/stages/SimpleImportsSolver.java
index 38dfa5b3..d5eb276d 100644
--- a/src/main/java/com/github/sommeri/less4j/core/compiler/stages/SimpleImportsSolver.java
+++ b/src/main/java/com/github/sommeri/less4j/core/compiler/stages/SimpleImportsSolver.java
@@ -2,0 +3 @@ package com.github.sommeri.less4j.core.compiler.stages;
+import java.io.File;
@@ -148 +149 @@ public class SimpleImportsSolver {
-    if (filename.toLowerCase().contains("."))
+    if ((new File(filename)).getName().contains("."))