diff --git a/optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/generic/SwapMove.java b/optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/generic/SwapMove.java
index c633154..87416a4 100644
--- a/optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/generic/SwapMove.java
+++ b/optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/generic/SwapMove.java
@@ -62,0 +63,3 @@ public class SwapMove extends AbstractMove {
+            if (leftEntity == null || rightEntity == null) {
+                return false;
+            }
