diff --git a/rio/src/main/java/org/semanticweb/owlapi/formats/BinaryRdfOntologyFormatFactory.java b/rio/src/main/java/org/semanticweb/owlapi/formats/BinaryRdfOntologyFormatFactory.java
index 54602faaa..d9183b72f 100644
--- a/rio/src/main/java/org/semanticweb/owlapi/formats/BinaryRdfOntologyFormatFactory.java
+++ b/rio/src/main/java/org/semanticweb/owlapi/formats/BinaryRdfOntologyFormatFactory.java
@@ -17,3 +17 @@ public class BinaryRdfOntologyFormatFactory extends
-    public BinaryRdfOntologyFormatFactory() {
-        super();
-    }
+    private static final long serialVersionUID = 40000L;