diff --git a/hikaricp/src/main/java/com/zaxxer/hikari/HikariDataSource.java b/hikaricp/src/main/java/com/zaxxer/hikari/HikariDataSource.java
index f55aa0f..40319ab 100644
--- a/hikaricp/src/main/java/com/zaxxer/hikari/HikariDataSource.java
+++ b/hikaricp/src/main/java/com/zaxxer/hikari/HikariDataSource.java
@@ -276 +276 @@ public class HikariDataSource extends HikariConfig implements DataSource, Closea
-         else if (!username.equals(otherKey.username)) {
+         else if (username != otherKey.username) {
@@ -282 +282 @@ public class HikariDataSource extends HikariConfig implements DataSource, Closea
-         else if (!password.equals(otherKey.password)) {
+         else if (password != otherKey.password) {
