diff --git a/modules/cpr/src/main/java/org/atmosphere/cpr/AtmosphereResourceFactory.java b/modules/cpr/src/main/java/org/atmosphere/cpr/AtmosphereResourceFactory.java
index f3d0f84..f594855 100644
--- a/modules/cpr/src/main/java/org/atmosphere/cpr/AtmosphereResourceFactory.java
+++ b/modules/cpr/src/main/java/org/atmosphere/cpr/AtmosphereResourceFactory.java
@@ -196,0 +197 @@ public final class AtmosphereResourceFactory {
+        if (uuid == null) return null;
