diff --git a/src/net/sourceforge/plantuml/graphic/Img.java b/src/net/sourceforge/plantuml/graphic/Img.java
index a6a8bf1b..0b653664 100644
--- a/src/net/sourceforge/plantuml/graphic/Img.java
+++ b/src/net/sourceforge/plantuml/graphic/Img.java
@@ -31 +31 @@
- * Revision $Revision: 14056 $
+ * Revision $Revision: 14726 $
@@ -49,0 +50 @@ import net.sourceforge.plantuml.FileSystem;
+import net.sourceforge.plantuml.StringUtils;
@@ -78 +79 @@ public class Img implements HtmlCommand {
-		return ImgValign.valueOf(m.group(1).toUpperCase());
+		return ImgValign.valueOf(StringUtils.goUpperCase(m.group(1)));