diff --git a/src/main/java/com/tdunning/math/stats/ArrayDigest.java b/src/main/java/com/tdunning/math/stats/ArrayDigest.java
index 9591332..3bcc2b7 100644
--- a/src/main/java/com/tdunning/math/stats/ArrayDigest.java
+++ b/src/main/java/com/tdunning/math/stats/ArrayDigest.java
@@ -824,0 +825,2 @@ public class ArrayDigest extends AbstractTDigest {
+            assert active == pageSize;
+            final int half = pageSize / 2;
@@ -826,2 +828,2 @@ public class ArrayDigest extends AbstractTDigest {
-            System.arraycopy(centroids, 16, newPage.centroids, 0, pageSize / 2);
-            System.arraycopy(counts, 16, newPage.counts, 0, pageSize / 2);
+            System.arraycopy(centroids, half, newPage.centroids, 0, pageSize - half);
+            System.arraycopy(counts, half, newPage.counts, 0, pageSize - half);
@@ -830 +832 @@ public class ArrayDigest extends AbstractTDigest {
-                newPage.history.addAll(history.subList(pageSize / 2, pageSize));
+                newPage.history.addAll(history.subList(half, pageSize));
@@ -833 +835 @@ public class ArrayDigest extends AbstractTDigest {
-                tmp.addAll(history.subList(0, pageSize / 2));
+                tmp.addAll(history.subList(0, half));
@@ -836,2 +838,2 @@ public class ArrayDigest extends AbstractTDigest {
-            active = 16;
-            newPage.active = 16;
+            active = half;
+            newPage.active = pageSize - half;
@@ -841 +843 @@ public class ArrayDigest extends AbstractTDigest {
-            for (int i = 0; i < 16; i++) {
+            for (int i = 0; i < half; i++) {
