diff --git a/parsers/src/main/java/org/coode/owl/krssparser/KRSSOntologyFormat.java b/parsers/src/main/java/org/coode/owl/krssparser/KRSSOntologyFormat.java
index 34f2d0530..adb451edd 100644
--- a/parsers/src/main/java/org/coode/owl/krssparser/KRSSOntologyFormat.java
+++ b/parsers/src/main/java/org/coode/owl/krssparser/KRSSOntologyFormat.java
@@ -51,0 +52 @@ public class KRSSOntologyFormat extends OWLOntologyFormat {
+    private static final long serialVersionUID = 30405L;