diff --git a/logback-classic/src/main/java/ch/qos/logback/classic/jmx/JMXConfigurator.java b/logback-classic/src/main/java/ch/qos/logback/classic/jmx/JMXConfigurator.java
index 9f241c3..c6a4be2 100644
--- a/logback-classic/src/main/java/ch/qos/logback/classic/jmx/JMXConfigurator.java
+++ b/logback-classic/src/main/java/ch/qos/logback/classic/jmx/JMXConfigurator.java
@@ -3 +3 @@
- * Copyright (C) 1999-2015, QOS.ch. All rights reserved.
+ * Copyright (C) 1999-2016, QOS.ch. All rights reserved.
@@ -142,0 +143,3 @@ public class JMXConfigurator extends ContextAwareBase implements
+
+    // use configurator only if a url is defined
+    if (url != null) {
@@ -157,0 +161 @@ public class JMXConfigurator extends ContextAwareBase implements
+  }
