diff --git a/biojava3-structure/src/main/java/org/biojava/bio/structure/PDBCrystallographicInfo.java b/biojava3-structure/src/main/java/org/biojava/bio/structure/PDBCrystallographicInfo.java
index 090288bff..e55376f30 100644
--- a/biojava3-structure/src/main/java/org/biojava/bio/structure/PDBCrystallographicInfo.java
+++ b/biojava3-structure/src/main/java/org/biojava/bio/structure/PDBCrystallographicInfo.java
@@ -14,0 +15,4 @@ public class PDBCrystallographicInfo implements Serializable {
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = -7949886749566087669L;