diff --git a/maven-model-builder/src/main/java/org/apache/maven/model/composition/DefaultDependencyManagementImporter.java b/maven-model-builder/src/main/java/org/apache/maven/model/composition/DefaultDependencyManagementImporter.java
index 0854477..831e754 100644
--- a/maven-model-builder/src/main/java/org/apache/maven/model/composition/DefaultDependencyManagementImporter.java
+++ b/maven-model-builder/src/main/java/org/apache/maven/model/composition/DefaultDependencyManagementImporter.java
@@ -110 +110 @@ public void importManagement( final Model target, final List<? extends Dependenc
-                                "Multiple conflicting imports of dependency '%1$s' into model '%2$s' @ '%3$s' (%4$s). "
+                                "Multiple conflicting imports of dependency '%1$s' into model '%2$s'%3$s(%4$s). "
@@ -117 +117,3 @@ public void importManagement( final Model target, final List<? extends Dependenc
-                                target.getPomFile().getAbsolutePath(), conflictsBuilder.substring( 2 ) ) ) );
+                                target.getPomFile() != null
+                                    ? " @ '" + target.getPomFile().getAbsolutePath() + "' "
+                                    : " ", conflictsBuilder.substring( 2 ) ) ) );
