diff --git a/modules/cpr/src/main/java/org/atmosphere/cpr/AtmosphereResourceFactory.java b/modules/cpr/src/main/java/org/atmosphere/cpr/AtmosphereResourceFactory.java
index 5bd473d..91c9b08 100644
--- a/modules/cpr/src/main/java/org/atmosphere/cpr/AtmosphereResourceFactory.java
+++ b/modules/cpr/src/main/java/org/atmosphere/cpr/AtmosphereResourceFactory.java
@@ -47,0 +48 @@ public final class AtmosphereResourceFactory {
+                            if (method.getName().equals("isDestroyed")) return false;
