diff --git a/src/main/java/hudson/remoting/Engine.java b/src/main/java/hudson/remoting/Engine.java
index 4bb2e257..6860ef30 100644
--- a/src/main/java/hudson/remoting/Engine.java
+++ b/src/main/java/hudson/remoting/Engine.java
@@ -165 +165 @@ public class Engine extends Thread {
-                        String encoding = Base64.encode(credentials.getBytes());
+                        String encoding = Base64.encode(credentials.getBytes("UTF-8"));