diff --git a/src/net/sourceforge/plantuml/Option.java b/src/net/sourceforge/plantuml/Option.java
index 33982587..3800a090 100644
--- a/src/net/sourceforge/plantuml/Option.java
+++ b/src/net/sourceforge/plantuml/Option.java
@@ -31 +31 @@
- * Revision $Revision: 14321 $
+ * Revision $Revision: 14727 $
@@ -243 +243 @@ public class Option {
-			} else if (s.toLowerCase().startsWith("-ftp")) {
+			} else if (StringUtils.goLowerCase(s).startsWith("-ftp")) {