diff --git a/modules/activiti-engine/src/main/java/org/activiti/engine/impl/webservice/WSOperation.java b/modules/activiti-engine/src/main/java/org/activiti/engine/impl/webservice/WSOperation.java
index 15f1474..6a814f4 100644
--- a/modules/activiti-engine/src/main/java/org/activiti/engine/impl/webservice/WSOperation.java
+++ b/modules/activiti-engine/src/main/java/org/activiti/engine/impl/webservice/WSOperation.java
@@ -17,0 +18 @@ import java.util.logging.Logger;
+import org.activiti.engine.impl.bpmn.MessageDefinition;
@@ -86 +87,4 @@ public class WSOperation implements OperationImplementation {
-    MessageInstance message = operation.getOutMessage().createInstance();
+    MessageInstance message = null;
+    MessageDefinition outMessage = operation.getOutMessage();
+    if (outMessage != null) {
+      message = outMessage.createInstance();
@@ -87,0 +92 @@ public class WSOperation implements OperationImplementation {
+    }
