diff --git a/modules/cpr/src/main/java/org/atmosphere/container/version/JSR356WebSocket.java b/modules/cpr/src/main/java/org/atmosphere/container/version/JSR356WebSocket.java
index 871b5c7..8c61752 100644
--- a/modules/cpr/src/main/java/org/atmosphere/container/version/JSR356WebSocket.java
+++ b/modules/cpr/src/main/java/org/atmosphere/container/version/JSR356WebSocket.java
@@ -162 +162 @@ public class JSR356WebSocket extends WebSocket {
-            if (!result.isOK() || result.getException() != null && r != null) {
+            if (!result.isOK() || result.getException() != null) {
@@ -163,0 +164 @@ public class JSR356WebSocket extends WebSocket {
+                if (r != null) {
@@ -169,0 +171 @@ public class JSR356WebSocket extends WebSocket {
+}
\ No newline at end of file
