diff --git a/src/main/java/org/aeonbits/owner/ConfigURLFactory.java b/src/main/java/org/aeonbits/owner/ConfigURLFactory.java
index 2f36e7f..0c6a46c 100644
--- a/src/main/java/org/aeonbits/owner/ConfigURLFactory.java
+++ b/src/main/java/org/aeonbits/owner/ConfigURLFactory.java
@@ -11 +10,0 @@ package org.aeonbits.owner;
-import java.io.Serializable;
@@ -18 +17 @@ import java.net.URL;
-class ConfigURLFactory implements Serializable {
+class ConfigURLFactory {
@@ -20 +19 @@ class ConfigURLFactory implements Serializable {
-    private transient final ClassLoader classLoader;
+    private final transient ClassLoader classLoader;