diff --git a/src/main/java/teetime/framework/pipe/PipeFactoryLoader.java b/src/main/java/teetime/framework/pipe/PipeFactoryLoader.java
index 2e129e2f..6e564014 100644
--- a/src/main/java/teetime/framework/pipe/PipeFactoryLoader.java
+++ b/src/main/java/teetime/framework/pipe/PipeFactoryLoader.java
@@ -17 +17 @@ import teetime.framework.FileSearcher;
-public class PipeFactoryLoader {
+public final class PipeFactoryLoader {
@@ -28 +28 @@ public class PipeFactoryLoader {
-		BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(stream));
+		BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(stream, "UTF-8"));