diff --git a/src/main/java/nl/inl/blacklab/externalstorage/ContentStoreDir.java b/src/main/java/nl/inl/blacklab/externalstorage/ContentStoreDir.java
index 2a4fe6f6..3dfa564a 100644
--- a/src/main/java/nl/inl/blacklab/externalstorage/ContentStoreDir.java
+++ b/src/main/java/nl/inl/blacklab/externalstorage/ContentStoreDir.java
@@ -23 +22,0 @@ import java.io.FileOutputStream;
-import java.io.FileReader;
@@ -25,0 +25 @@ import java.io.IOException;
+import java.io.InputStreamReader;
@@ -186 +186 @@ public class ContentStoreDir extends ContentStoreDirAbstract {
-			BufferedReader f = new BufferedReader(new FileReader(tocFile));
+			BufferedReader f = new BufferedReader(new InputStreamReader(new FileInputStream(tocFile), "utf-8"));
@@ -220 +220 @@ public class ContentStoreDir extends ContentStoreDirAbstract {
-			PrintWriter f = new PrintWriter(new FileWriter(tocFile));
+			PrintWriter f = new PrintWriter(new OutputStreamWriter(new FileOutputStream(tocFile), "utf-8"));