diff --git a/src/org/jgroups/protocols/pbcast/NAKACK2.java b/src/org/jgroups/protocols/pbcast/NAKACK2.java
index 6442f5e..e71a544 100644
--- a/src/org/jgroups/protocols/pbcast/NAKACK2.java
+++ b/src/org/jgroups/protocols/pbcast/NAKACK2.java
@@ -14,4 +14 @@ import java.util.*;
-import java.util.concurrent.ConcurrentMap;
-import java.util.concurrent.Executor;
-import java.util.concurrent.Future;
-import java.util.concurrent.TimeUnit;
+import java.util.concurrent.*;
@@ -176 +173 @@ public class NAKACK2 extends Protocol implements DiagnosticsHandler.ProbeHandler
-    protected final Map<Address,Long>   xmit_task_map=new HashMap<Address,Long>();
+    protected final Map<Address,Long>   xmit_task_map=new ConcurrentHashMap<Address,Long>();
