diff --git a/src/main/java/nodebox/client/NodeBoxDocument.java b/src/main/java/nodebox/client/NodeBoxDocument.java
index 5c0c4150..f0b1e844 100644
--- a/src/main/java/nodebox/client/NodeBoxDocument.java
+++ b/src/main/java/nodebox/client/NodeBoxDocument.java
@@ -29,2 +29 @@ import java.io.StringWriter;
-import java.util.ArrayList;
-import java.util.HashMap;
+import java.util.*;
@@ -32 +30,0 @@ import java.util.List;
-import java.util.Map;
@@ -1522 +1520 @@ public class NodeBoxDocument extends JFrame implements WindowListener, HandleDel
-        if (file.getName().toLowerCase().endsWith(".pdf"))
+        if (file.getName().toLowerCase(Locale.US).endsWith(".pdf"))