diff --git a/src/net/sourceforge/plantuml/classdiagram/command/JavaFile.java b/src/net/sourceforge/plantuml/classdiagram/command/JavaFile.java
index d17daafa..9953239b 100644
--- a/src/net/sourceforge/plantuml/classdiagram/command/JavaFile.java
+++ b/src/net/sourceforge/plantuml/classdiagram/command/JavaFile.java
@@ -31 +31 @@
- * Revision $Revision: 12235 $
+ * Revision $Revision: 14726 $
@@ -45,0 +46 @@ import java.util.regex.Pattern;
+import net.sourceforge.plantuml.StringUtils;
@@ -83 +84 @@ class JavaFile {
-					final LeafType type = LeafType.valueOf(matchClassDefinition.group(1).toUpperCase());
+					final LeafType type = LeafType.valueOf(StringUtils.goUpperCase(matchClassDefinition.group(1)));