diff --git a/src/net/sourceforge/plantuml/sequencediagram/command/CommandActivate.java b/src/net/sourceforge/plantuml/sequencediagram/command/CommandActivate.java
index e826989f..69e08b6d 100644
--- a/src/net/sourceforge/plantuml/sequencediagram/command/CommandActivate.java
+++ b/src/net/sourceforge/plantuml/sequencediagram/command/CommandActivate.java
@@ -31 +31 @@
- * Revision $Revision: 14321 $
+ * Revision $Revision: 14726 $
@@ -53 +53 @@ public class CommandActivate extends SingleLineCommand<SequenceDiagram> {
-		final LifeEventType type = LifeEventType.valueOf(arg.get(0).toUpperCase());
+		final LifeEventType type = LifeEventType.valueOf(StringUtils.goUpperCase(arg.get(0)));