diff --git a/org.eclipse.jgit/src/org/eclipse/jgit/api/ReflogCommand.java b/org.eclipse.jgit/src/org/eclipse/jgit/api/ReflogCommand.java
index 8cd78ae..4536af1 100644
--- a/org.eclipse.jgit/src/org/eclipse/jgit/api/ReflogCommand.java
+++ b/org.eclipse.jgit/src/org/eclipse/jgit/api/ReflogCommand.java
@@ -50,0 +51 @@
+import org.eclipse.jgit.api.errors.RefNotFoundException;
@@ -99,0 +101,3 @@ public ReflogCommand setRef(String ref) {
+			if (reader == null)
+				throw new RefNotFoundException(MessageFormat.format(
+						JGitText.get().refNotResolved, ref));
