diff --git a/src/main/java/fr/ens/transcriptome/eoulsan/steps/mapping/hadoop/SAMFilterReducer.java b/src/main/java/fr/ens/transcriptome/eoulsan/steps/mapping/hadoop/SAMFilterReducer.java
index 030bf484..9f66f20f 100644
--- a/src/main/java/fr/ens/transcriptome/eoulsan/steps/mapping/hadoop/SAMFilterReducer.java
+++ b/src/main/java/fr/ens/transcriptome/eoulsan/steps/mapping/hadoop/SAMFilterReducer.java
@@ -164 +164,2 @@ public class SAMFilterReducer extends Reducer<Text, Text, Text, Text> {
-            new BufferedReader(new InputStreamReader(fs.open(bestFile)));
+            new BufferedReader(new InputStreamReader(fs.open(bestFile),
+                "ISO-8859-1"));